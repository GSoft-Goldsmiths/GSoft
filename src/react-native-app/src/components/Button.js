import React from 'react';
import { style } from '../styles/variables';
import { Text, TouchableHighlight } from 'react-native';

export default ({ text, handler }) => (
  <TouchableHighlight
    style={style.button}
    onPress={() => handler()}
  >
    <Text style={{ color: '#fff' }}>{text}</Text>
  </TouchableHighlight>
);