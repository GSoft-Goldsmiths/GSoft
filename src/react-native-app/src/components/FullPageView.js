import React from 'react';
import { fullPageView, style } from '../styles/variables';
import Button from './Button';
import { Image, Text, View } from 'react-native';

export default ({
                  containerStyle,
                  imageStyle,
                  imageSource,
                  title,
                  subtitle,
                  buttonText,
                  buttonHandler,
                }) => {

  return (
    <View style={style.container}>
      <Image
        style={fullPageView.image}
        source={imageSource}
      />
      <Text style={fullPageView.title}>{title}</Text>
      {subtitle ? <Text style={fullPageView.subtitle}>{subtitle}</Text> : null}
      <Button text={buttonText} handler={buttonHandler}/>
    </View>
  );
}