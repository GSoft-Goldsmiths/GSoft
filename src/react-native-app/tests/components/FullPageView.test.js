import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import { Image, Text, View } from 'react-native';

import FullPageView from '../../src/components/FullPageView';
import Button from '../../src/components/Button';

configure({ adapter: new Adapter() });

describe('<FullPageView />', ()=>{
  
  it('renders without crashing', () => {
    const rendered = renderer.create(<FullPageView text="title"  buttonHandler={()=>{}} buttonText="button"/>).
      toJSON();
    expect(rendered).toMatchSnapshot();
  });
  
  it('renders correct elements', () => {
    const wrapper = shallow(<FullPageView text="title" subtitle="subtitle" buttonHandler={()=>{}} buttonText="button"/>);
    expect(wrapper.find(View).exists()).toBeTruthy();
    expect(wrapper.find(View).length).toEqual(1);
    expect(wrapper.find(Image).exists()).toBeTruthy();
    expect(wrapper.find(Image).length).toEqual(1);
    expect(wrapper.find(Text).exists()).toBeTruthy();
    expect(wrapper.find(Text).length).toEqual(2);
    expect(wrapper.find(Button).exists()).toBeTruthy();
    expect(wrapper.find(Button).length).toEqual(1);
  });
});

