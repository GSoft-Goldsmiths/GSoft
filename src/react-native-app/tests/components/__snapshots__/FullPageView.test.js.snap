// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullPageView /> renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <Image
    source={undefined}
    style={
      Object {
        "height": 145,
        "marginBottom": 20,
        "width": 145,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#0093A8",
        "fontFamily": "Caviar Dreams",
        "fontSize": 30,
        "marginBottom": 15,
      }
    }
  />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0093A8",
        "height": 36,
        "justifyContent": "center",
        "marginVertical": 20,
        "width": 90,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Caviar Dreams Bold",
        }
      }
    >
      button
    </Text>
  </View>
</View>
`;
